spec/features/find/search/undergraduate/course_results_spec.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+feature 'Questions and results for undergraduate courses' do
+  after do
+    Timecop.return
+  end
+
+  scenario 'in the 2025 cycle with the TDA feature active and searching for secondary courses' do
+    given_i_have_2025_courses
+    and_i_am_in_the_2025_cycle
+    and_the_tda_feature_flag_is_active
+    when_i_visit_the_start_page
+    and_i_select_the_across_england_radio_button
+    and_i_click_continue
+    and_i_choose_secondary
+    and_i_click_continue
+    and_i_choose_subjects
+    and_i_click_continue
+    then_i_am_on_the_degree_question_page
+
+    and_i_click_continue
+    then_i_see_an_error_message_on_the_degree_question_page
+    and_the_back_link_points_to_the_secondary_subjects_page
+
+    when_i_choose_no_i_do_not_have_a_degree
+    and_i_click_continue
+    then_i_am_on_the_visa_status_page
+    and_the_back_link_points_to_the_degree_question
+
+    when_i_choose_yes
+    and_i_click_find_courses
+    then_i_am_on_an_exit_page_for_no_degree_and_requires_visa_sponsorship
+
+    when_i_click_back
+    then_i_am_on_the_visa_status_page
+    and_the_back_link_points_to_the_degree_question
+
+    when_i_choose_no
+    and_i_click_find_courses
+
+    then_i_am_on_results_page
+    and_some_filters_are_hidden_for_undergraduate_courses
+    and_some_filters_are_visible_for_undergraduate_courses
+    and_i_only_see_secondary_undergraduate_courses
```

Comment:
Can we add a scenario where we select that Yes, we have an undergraduate degree? Should we see TDA courses?

Reply:
I did more below.

Reply:
I did more below but the scenario description was wrong. I fixed.